File: workflow-py/fetch_model.py
"""Download a compact ImageNet ONNX classifier used at task runtime."""

from __future__ import annotations

from pathlib import Path
from urllib.request import urlretrieve

MODEL_URL = (
    "https://github.com/onnx/models/raw/main/validated/vision/classification/"
    "squeezenet/model/squeezenet1.1-7.onnx"
)
MODEL_PATH = Path(__file__).parent / "models" / "squeezenet1.1-7.onnx"


def main() -> None:
    """Fetch the ONNX weights if they are not already on disk."""
    MODEL_PATH.parent.mkdir(parents=True, exist_ok=True)
    if MODEL_PATH.exists() and MODEL_PATH.stat().st_size > 1_000_000:
        print(f"model already present: {MODEL_PATH}")
        return
    print(f"downloading {MODEL_URL}")
    urlretrieve(MODEL_URL, MODEL_PATH)
    print(f"saved {MODEL_PATH} ({MODEL_PATH.stat().st_size} bytes)")


if __name__ == "__main__":
    main()

File: workflow-ts/fetch-model.mjs
import { mkdir, stat, writeFile } from "node:fs/promises";
import { dirname, join } from "node:path";
import { fileURLToPath } from "node:url";

const MODEL_URL =
  "https://github.com/onnx/models/raw/main/validated/vision/classification/squeezenet/model/squeezenet1.1-7.onnx";

const root = dirname(fileURLToPath(import.meta.url));
const modelPath = join(root, "models", "squeezenet1.1-7.onnx");

await mkdir(dirname(modelPath), { recursive: true });

try {
  const existing = await stat(modelPath);
  if (existing.size > 1_000_000) {
    console.log(`model already present: ${modelPath}`);
    process.exit(0);
  }
} catch {
  // download below
}

console.log(`downloading ${MODEL_URL}`);
const response = await fetch(MODEL_URL, { redirect: "follow" });
if (!response.ok) {
  throw new Error(`download failed: ${response.status} ${response.statusText}`);
}
const bytes = Buffer.from(await response.arrayBuffer());
await writeFile(modelPath, bytes);
console.log(`saved ${modelPath} (${bytes.length} bytes)`);
